File: src/main/java/com/kelsos/mbrc/commands/model/UpdateVolume.java
package com.kelsos.mbrc.commands.model;

import com.google.inject.Inject;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.model.MainDataModel;
import org.codehaus.jackson.node.IntNode;

public class UpdateVolume implements ICommand {
    @Inject
    MainDataModel model;
    @Override
    public void execute(IEvent e) {
        model.setVolume(((IntNode)e.getData()).asInt());
    }
}

File: app/src/main/java/com/kelsos/mbrc/commands/model/UpdateVolume.kt
package com.kelsos.mbrc.commands.model

import com.fasterxml.jackson.databind.node.IntNode
import com.kelsos.mbrc.events.bus.RxBus
import com.kelsos.mbrc.events.ui.VolumeChange
import com.kelsos.mbrc.interfaces.ICommand
import com.kelsos.mbrc.interfaces.IEvent
import com.kelsos.mbrc.model.MainDataModel
import javax.inject.Inject

class UpdateVolume
  @Inject
  constructor(
    private val model: MainDataModel,
    private val bus: RxBus,
  ) : ICommand {
    override fun execute(e: IEvent) {
      val newVolume = (e.data as IntNode).asInt()
      if (newVolume != model.volume) {
        model.volume = newVolume
        bus.post(VolumeChange(model.volume))
      }
    }
  }
